### DISCUSSION TITLE
CIE XYZ tristimulus values from spectral distribution are different between "colour" and manually calculated values.

### DISCUSSION THREAD

**Discussion by leklbk** [colour-science/colour#729] [Q&A] [answered]
I calculated XYZ from 1nm-interval wavelength with simple integration and comparing with values from colour science package. they were different.

I also compared these values with spectral measurement system which shows luminance and this luminance value same with the calculated value from integration

I also tried to change methods and CMFS in sd_to_XYZ but it didn't make significant change.

could you let me know why they are different?

**Comment by thomasmansencal** [comment]
Hi @leklbk,

Without seeing your code it will be tricky. How much different it is? Our computations should be correct and are verified against various CIE and ASTM publications.

Cheers,

Thomas

**Comment by leklbk** [comment]
Hi @KelSolaar , 
First of all, I really appreciate that I can use this unbelievable powerful tool.
I can't attach the file because of security policy of my company but I'm gonna try to explain as much as I can

I read oled spectrum measured data that has wavelength interval of 2nm
I downloaded color matching function from CIE technical site 1931, 2degree CMF and
just interpolate these CMFs into 2nm interval data. and I did mensuration by parts
Y is about 3.221 and corresponding luminance was 2200 cd/m2 which is correspond to measured luminance (CAS-140)

I also tried to do the same process using colour package, and I interpolated spectral data to 1nm, 5nm, 10nm intervals. convert sd_to_XYZ and Y*683
luminance was 2058, 1936, 2507 for each cases. 

I think interpolated spectra by 5nm, 10nm could be different because this spectrum has quite strong interference pattern. 

However, I couldn't explain why luminace from 1nm interpolated spectrum data and measured data are different.
I also checked  interpolation method of CMFs but it doesn't change a lot (Lineaer, Sprague).
I also changed sd_to_XYZ method integration and ASTM but it doesn't change a lot, too.

If you need specific information, I will try to find a way to comply with our security policy

Thank you.

**Comment by thomasmansencal** [comment]
Interesting, feel like your spectrum is either very noisy or has a lot of peaks, have you tried instead of converting the CMFs to 2nm to actually interpolate your spectrum to 1nm? Would be great to get a mock spectrum for me to test, if you are allowed, you can send it to me via email.

Cheers,

**Comment by thomasmansencal** [comment, marked as answer]
Closing this one as it is handled privately!
